File: maps/src/main/java/it/localhostsoftware/maps/google/model/GoogleIndoorBuilding.java
package it.localhostsoftware.maps.google.model;

import java.util.ArrayList;
import java.util.List;

import it.localhostsoftware.maps.model.IndoorBuilding;
import it.localhostsoftware.maps.model.IndoorLevel;

public class GoogleIndoorBuilding extends IndoorBuilding<com.google.android.gms.maps.model.IndoorBuilding> {
    public GoogleIndoorBuilding(com.google.android.gms.maps.model.IndoorBuilding IB) {
        super(IB);
    }

    @Override
    public int getDefaultLevelIndex() {
        return getIndoorBuilding().getDefaultLevelIndex();
    }

    @Override
    public int getActiveLevelIndex() {
        return getIndoorBuilding().getActiveLevelIndex();
    }

    @Override
    public List<IndoorLevel<?>> getLevels() {
        ArrayList<IndoorLevel<?>> out = new ArrayList<>(getIndoorBuilding().getLevels().size());
        for (com.google.android.gms.maps.model.IndoorLevel value : getIndoorBuilding().getLevels())
            out.add(new GoogleIndoorLevel(value));
        return out;
    }

    @Override
    public boolean isUnderground() {
        return getIndoorBuilding().isUnderground();
    }

    @Override
    public boolean equals(Object var1) {
        return getIndoorBuilding().equals(var1);
    }

    @Override
    public int hashCode() {
        return getIndoorBuilding().hashCode();
    }
}

File: maps/src/main/java/it/localhostsoftware/maps/google/model/GoogleIndoorBuilding.kt
package it.localhostsoftware.maps.google.model

import it.localhostsoftware.maps.model.IndoorBuilding

class GoogleIndoorBuilding(ib: com.google.android.gms.maps.model.IndoorBuilding) : IndoorBuilding<com.google.android.gms.maps.model.IndoorBuilding, GoogleIndoorLevel>(ib) {
    override val defaultLevelIndex: Int
        get() = ib.defaultLevelIndex
    override val activeLevelIndex: Int
        get() = ib.activeLevelIndex
    override val levels: List<GoogleIndoorLevel>
        get() = ib.levels.map { GoogleIndoorLevel(it) }
    override val isUnderground: Boolean
        get() = ib.isUnderground

    override fun equals(other: Any?): Boolean = ib == other
    override fun hashCode(): Int = ib.hashCode()
}